kratos/utilities/math_utils.h

```diff
@@ -941,6 +941,21 @@ class MathUtils
         c/=norm;
     }
     
+    /**
+     * Computes the angle between two vectors in 3D
+     * @param v1 First input vector
+     * @param v2 Second input vector
+     */
+    
+    template< class T1, class T2>
```

Comment:
I THINK that your formula fails if the two vectors are e.g.

v1 = (1,0,0)
v2 = (-1,0,0)

take a look at this thread
https://stackoverflow.com/questions/10133957/signed-angle-between-two-vectors-without-a-reference-plane

(note that you cannot assign an angle between the two vectors unless you assign an orientation)

Reply:
This would explain some errors I am getting

Reply:
the point is that

      inner_prod(v1,v2) = norm(v1)*norm(v2)*cos(theta)
      norm(cross_prod(v1,v2)) =  norm(v1)*norm(v2)*sin(theta)

so that

    tan(theta)  = norm(cross_prod(v1,v2))/inner_prod(v1,v2)

I would implement this as

        array_1d<double,3> n;
        MathUtils<double>::CrossProduct(n,v1,v2);
        const double numerator = norm_2(n);
        const double denominator = inner_prod(v1,v2);
        const double angle = atan2(numerator, denominator);

having said this, the possibility of obtaining the cos instead the angle itself as proposed by @msandre is definitely worth considering

Reply:
https://people.eecs.berkeley.edu/~wkahan/Mindless.pdf page 47, from https://es.mathworks.com/matlabcentral/answers/16243-angle-between-two-vectors-in-3d

Reply:
ok the first article is convincing to me. i am accepting your solution (and thx for being didactic)

Reply:
I tried :kissing_smiling_eyes: